### PR TITLE
Add Sparse Connection support to mysqli

### PR DESCRIPTION
**Pull request by tyleruebele**
If instantiated with sparse = true, will close the connection to mysql server after each function call and reopen for the next.  Stores properties between connections.

### CODE REVIEW

File: src/data/mysqli_.php [hunk 1]
```diff
@@ -105,10 +235,11 @@ public function __call($name, $arguments) {
      * @return void
      */
     public function close() {
-        if ($this->_open) {
+        if ($this->_inner_is_open) {
             $this->_mysqli->close();
-            $this->_open = false;
+            $this->_inner_is_open = false;
         }
```
Comment (markneumann99) [inline src/data/mysqli_.php:241]: This looks a lot like `_inner_close`. But maybe no reason to go through that.

File: src/data/mysqli_.php [hunk 2]
```diff
@@ -233,10 +364,11 @@ public function query($query, $resultMode = \MYSQLI_STORE_RESULT) {
...
-        // Handle "MySQL server has gone away"
+        // Handle "MySQL server has gone away" by reopening the inner connection
         if (2006 == $this->_mysqli->errno) {
-            $this->_mysqli = new mysqli(...$this->_connectionParam);
-            if (!mysqli_connect_error()) {
+            $this->_inner_close(true);
+            $this->_inner_open();
+            if ($this->_inner_is_open) {
                 $result = $this->_mysqli->query($query, $resultmode);
             }
```
Comment (markneumann99) [inline src/data/mysqli_.php:373]: should have an `else { trigger_error('Refusing to run a query against a closed connection.'); }` ??

**Review by andyleach — APPROVED** [review]

File: src/data/mysqli_.php (hunk 1 above)
Comment (tyleruebele) [reply src/data/mysqli_.php:241]: They are just different enough.

File: src/data/mysqli_.php (hunk 2 above)
Comment (tyleruebele) [reply src/data/mysqli_.php:373]: Fixup commit ensures this is only called by `_inner_query` which emits that error.

**Review by markneumann99 — APPROVED** [review]

**Pull request merged**